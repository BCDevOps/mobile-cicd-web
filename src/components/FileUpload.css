.file-upload-container {
  background-color: white;
  /* margin: 0em 3em 0em 3em; */
}

.file-upload-container .drop-zone {
  text-align: center;
  height: 3em;
  /* width: 16em; */
  background-color: white;
  border-style: dashed;
  border-width: 1px;
  border-color: lightgray;
  border-radius: 0.5em;
  color: darkgrey;
  /* display: inline-block; */
  /* padding: 1.2em 1.2em 1.2em 1.2em; */
}

.file-icon {
  color: #fcba19;
}

.file-upload-container:disabled {
  background-color: lightcoral;
}

.file-upload-container ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  text-align: left;
}

.file-upload-container .title {
  font-size: 1.3em;
}

.file-upload-container .sub-title {
  font-size: 0.8em;
}
