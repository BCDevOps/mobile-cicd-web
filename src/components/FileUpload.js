//
// DevHub
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Jason Leach on 2018-09-04.
//

import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import { addFile } from '../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './FileUpload.css';

class FileUpload extends Component {
  onDrop = async (acceptedFiles, rejectedFiles) => {
    console.log(this.props);
    console.log('Hello World', acceptedFiles);
    // const req = request.post('/upload');
    acceptedFiles.forEach(element => {
      console.log(`got file = ${element.name}`);
      this.props.addFile(element);
    });
  };

  size = sizeInBytes => {
    return (
      <div>
        &nbsp;&nbsp;
        {Math.round(sizeInBytes * 0.000001)}
        Mb
      </div>
    );
  };

  render() {
    return (
      <div className="file-upload-container">
        <Dropzone className="drop-zone" onDrop={files => this.onDrop(files)}>
          <div className="title">Drag file to upload.</div>
          <div className="sub-title">(.ZIP format only)</div>
        </Dropzone>
        <ul>
          {this.props.files.map(t => (
            <li key={t.name}>
              {<FontAwesomeIcon icon="file" className="file-icon" />}
              {t.name} {this.size(t.size)}
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    files: state.files,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      addFile,
    },
    dispatch
  );
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FileUpload);
