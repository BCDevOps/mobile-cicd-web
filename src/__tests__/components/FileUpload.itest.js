import React from 'react';
import { shallow } from 'enzyme';
import FileUpload from '../../components/FileUpload/FileUpload';

describe('FileUpload Component', () => {
  const onFileAccepted = jest.fn();
  const files = [{ name: 'file1', size: 100 }];
  const files2 = [{ name: 'file1', size: 100 }, { name: 'file2', size: 200 }];

  it('matches snapshot', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={files} />);
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('handles more than one files', () => {
    const wrapper = shallow(<FileUpload onFileAccepted={onFileAccepted} files={files2} />);
    expect(wrapper.find('.drop-zone').prop('disabled')).toBe(true);
    expect(wrapper.find('.title').text()).toBe('Only one file can be selected.');
    expect(wrapper.find('.sub-title').text()).toBe('');
  });
});
