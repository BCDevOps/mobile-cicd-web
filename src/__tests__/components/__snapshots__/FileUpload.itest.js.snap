// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FileUpload
    files={
      Array [
        Object {
          "name": "file1",
          "size": 100,
        },
      ]
    }
    onFileAccepted={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <n
          className="drop-zone"
          disableClick={false}
          disabled={true}
          getDataTransferItems={[Function]}
          inputProps={Object {}}
          maxSize={Infinity}
          minSize={0}
          multiple={true}
          onDrop={[MockFunction]}
          preventDropOnDocument={true}
        >
          <div
            className="title"
          >
            Only one file can be selected.
          </div>
          <div
            className="sub-title"
          />
        </n>,
        <ul
          className="file-list"
        >
          <li>
            <FontAwesomeIcon
              border={false}
              className="file-icon"
              fixedWidth={false}
              flip={null}
              icon="file"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span
              className="file-name"
            >
              file1
            </span>
            <span
              className="file-size"
            >
              <div>
                  
                0
                Mb
              </div>
            </span>
          </li>
        </ul>,
      ],
      "className": "file-upload-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="title"
            >
              Only one file can be selected.
            </div>,
            <div
              className="sub-title"
            />,
          ],
          "className": "drop-zone",
          "disableClick": false,
          "disabled": true,
          "getDataTransferItems": [Function],
          "inputProps": Object {},
          "maxSize": Infinity,
          "minSize": 0,
          "multiple": true,
          "onDrop": [MockFunction],
          "preventDropOnDocument": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Only one file can be selected.",
              "className": "title",
            },
            "ref": null,
            "rendered": "Only one file can be selected.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "sub-title",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <FontAwesomeIcon
                border={false}
                className="file-icon"
                fixedWidth={false}
                flip={null}
                icon="file"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              <span
                className="file-name"
              >
                file1
              </span>
              <span
                className="file-size"
              >
                <div>
                    
                  0
                  Mb
                </div>
              </span>
            </li>,
          ],
          "className": "file-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "file1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className="file-icon"
                  fixedWidth={false}
                  flip={null}
                  icon="file"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                <span
                  className="file-name"
                >
                  file1
                </span>,
                <span
                  className="file-size"
                >
                  <div>
                      
                    0
                    Mb
                  </div>
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "file-icon",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "file",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "file1",
                  "className": "file-name",
                },
                "ref": null,
                "rendered": "file1",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div>
                      
                    0
                    Mb
                  </div>,
                  "className": "file-size",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "  ",
                      0,
                      "Mb",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "  ",
                    0,
                    "Mb",
                  ],
                  "type": "div",
                },
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <n
            className="drop-zone"
            disableClick={false}
            disabled={true}
            getDataTransferItems={[Function]}
            inputProps={Object {}}
            maxSize={Infinity}
            minSize={0}
            multiple={true}
            onDrop={[MockFunction]}
            preventDropOnDocument={true}
          >
            <div
              className="title"
            >
              Only one file can be selected.
            </div>
            <div
              className="sub-title"
            />
          </n>,
          <ul
            className="file-list"
          >
            <li>
              <FontAwesomeIcon
                border={false}
                className="file-icon"
                fixedWidth={false}
                flip={null}
                icon="file"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              <span
                className="file-name"
              >
                file1
              </span>
              <span
                className="file-size"
              >
                <div>
                    
                  0
                  Mb
                </div>
              </span>
            </li>
          </ul>,
        ],
        "className": "file-upload-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                Only one file can be selected.
              </div>,
              <div
                className="sub-title"
              />,
            ],
            "className": "drop-zone",
            "disableClick": false,
            "disabled": true,
            "getDataTransferItems": [Function],
            "inputProps": Object {},
            "maxSize": Infinity,
            "minSize": 0,
            "multiple": true,
            "onDrop": [MockFunction],
            "preventDropOnDocument": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Only one file can be selected.",
                "className": "title",
              },
              "ref": null,
              "rendered": "Only one file can be selected.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sub-title",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <FontAwesomeIcon
                  border={false}
                  className="file-icon"
                  fixedWidth={false}
                  flip={null}
                  icon="file"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />
                <span
                  className="file-name"
                >
                  file1
                </span>
                <span
                  className="file-size"
                >
                  <div>
                      
                    0
                    Mb
                  </div>
                </span>
              </li>,
            ],
            "className": "file-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "file1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FontAwesomeIcon
                    border={false}
                    className="file-icon"
                    fixedWidth={false}
                    flip={null}
                    icon="file"
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  />,
                  <span
                    className="file-name"
                  >
                    file1
                  </span>,
                  <span
                    className="file-size"
                  >
                    <div>
                        
                      0
                      Mb
                    </div>
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "border": false,
                    "className": "file-icon",
                    "fixedWidth": false,
                    "flip": null,
                    "icon": "file",
                    "inverse": false,
                    "listItem": false,
                    "mask": null,
                    "pull": null,
                    "pulse": false,
                    "rotation": null,
                    "size": null,
                    "spin": false,
                    "symbol": false,
                    "title": "",
                    "transform": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "file1",
                    "className": "file-name",
                  },
                  "ref": null,
                  "rendered": "file1",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div>
                        
                      0
                      Mb
                    </div>,
                    "className": "file-size",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "  ",
                        0,
                        "Mb",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "  ",
                      0,
                      "Mb",
                    ],
                    "type": "div",
                  },
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
